### PR TITLE
Added recipe for Genio 3.0

### PR DESCRIPTION
**Pull request by jackburton79**

### CODE REVIEW

**Comment by Begasus** [comment]
Thanks!

File: haiku-apps/genio/genio-3.0.0.recipe [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+PROVIDES="
+	genio${secondaryArchSuffix} = $portVersion
+	app:Genio = $portVersion
+	"
+REQUIRES="
+	haiku${secondaryArchSuffix}
+	gcc${secondaryArchSuffix}_syslibs_devel
+	cmd:clang >= 16
+	lib:libeditorconfig$secondaryArchSuffix
+	lib:liblexilla$secondaryArchSuffix >= 5.2.4
```
Comment (korli) [inline haiku-apps/genio/genio-3.0.0.recipe:35]: >= 5.2.4 shouldn't be needed.

File: haiku-apps/genio/genio-3.0.0.recipe [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	lib:libgit2$secondaryArchSuffix
+	lib:libyaml_cpp$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libeditorconfig
```
Comment (korli) [inline haiku-apps/genio/genio-3.0.0.recipe:42]: devel:libeditorconfig$secondaryArchSuffix

Comment (Begasus) [reply haiku-apps/genio/genio-3.0.0.recipe:42]: My bad, didn't check 32bit (failed build on 32bit buildmaster @jackburton79 )
https://build.haiku-os.org/buildmaster/master/x86_gcc2/?buildrunDir=6531&viewMode=expanded

**Comment by jackburton79** [comment]
Shall I make these changes myself ?

**Comment by Begasus** [comment]
Would be nice yes :)

File: haiku-apps/genio/genio-3.0.0.recipe (hunk 1 above)
Comment (jackburton79) [reply haiku-apps/genio/genio-3.0.0.recipe:35]: you mean I should write "> 5.2.4" or remove the check altogether ?

**Comment by korli** [comment]
Also technically the recipe name should be genio-3.0.recipe because the version is 3.0

**Comment by jackburton79** [comment]
> Would be nice yes :) Make sure you use a clean branch though. ;)

Did in #10698.

> Also technically the recipe name should be genio-3.0.recipe because the version is 3.0

I guess. I just followed previous recipe style, though.

**Comment by Begasus** [comment]
> I guess. I just followed previous recipe style, though.

The archive is also 3.0 (v3.0.tar.gz), changing this in the recipe name to 3.0 allows to use $portVersion there (no need for manual changes then) and use "v$portVersion.tar.gz"

**Pull request merged**